### PR TITLE
Compare Series from the same anchor

### PR DESCRIPTION
**Pull request by xinrong-databricks**
Enable Series.compare to compare Series from the same anchor 

Preivous behavior:
```
>>> kser = ks.Series([1, 2])
>>> kser.compare(kser)
Traceback (most recent call last):
...
AssertionError: We don't need to combine. `this` and `that` are same.

>>> kser.compare(kser + 1)
Traceback (most recent call last):
...
AssertionError: We don't need to combine. `this` and `that` are same.
```

Current behavior:
```
>>> kser = ks.Series([1, 2])
>>> kser.compare(kser)
Empty DataFrame
Columns: [self, other]
Index: []

>>> kser.compare(kser + 1)
   self  other
0     1      2
1     2      3
```

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -5620,9 +5620,21 @@ def compare(
...
-            raise ValueError("Can only compare identically-labeled Series objects")
-        combined = combine_frames(self.to_frame(), other.to_frame())
+        if same_anchor(self, other):
+            self_column_label = verify_temp_column_name(other.to_frame(), "__self_column__")
+            other_column_label = verify_temp_column_name(self.to_frame(), "__other_column__")
+            combined = DataFrame(
+                self._internal.with_new_columns(
+                    [self._internal.data_spark_columns[0]]
+                    + [other._internal.data_spark_columns[0]],
+                    column_labels=[self_column_label] + [other_column_label],
```
Comment (ueshin) [inline databricks/koalas/series.py:5630]: nit: we can use one list for each: `[self..., other...]` instead of `[self...] + [other...]`.

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

File: databricks/koalas/series.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: 👍  Modified!

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -5620,9 +5620,21 @@ def compare(
...
-        if not self.index.equals(other.index):
-            raise ValueError("Can only compare identically-labeled Series objects")
-        combined = combine_frames(self.to_frame(), other.to_frame())
+        if same_anchor(self, other):
+            self_column_label = verify_temp_column_name(other.to_frame(), "__self_column__")
+            other_column_label = verify_temp_column_name(self.to_frame(), "__other_column__")
+            combined = DataFrame(
+                self._internal.with_new_columns(
+                    [self._internal.data_spark_columns[0]]
+                    + [other._internal.data_spark_columns[0]],
```
Comment (ueshin) [inline databricks/koalas/series.py:5629]: nit: this line, too?

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**